processor/src/main/java/org/mapstruct/ap/internal/model/ForgedMethod.java

```diff
@@ -249,7 +249,7 @@ public Type getReturnType() {
         return thrownTypes;
     }
 
-    public ForgedMethodHistory getHistory() {
+    public MethodDescription getHistory() {
```

Comment:
maybe we should rename the method to `getDescription` as well?

Reply:
I'll do so... This is all very much WIP.. The only thing I did was press refactor. I sometimes use CI to quickly check. 

I'll assign you as reviewer as soon as it becomes substantial 😄